asynction/security.py

```diff
@@ -0,0 +1,367 @@
+
+
+def load_scope_validate_func(scheme: SecurityScheme) -> ScopeValidateFunc:
+    scope_validate_func = None
+    if scheme.x_scope_validate_func:
+        scope_validate_func = load_handler(scheme.x_scope_validate_func)
+
+    if not scope_validate_func:
+        scope_validate_func = validate_scopes
+
+    return scope_validate_func
+
+
+def load_basic_info_func(scheme: SecurityScheme) -> BasicInfoFunc:
+    if not scheme.x_basic_info_func:
+        raise SecurityException("Missing basic info func")
+    try:
+        return load_handler(scheme.x_basic_info_func)
+    except ValueError as err:
+        raise SecurityException from err
+
+
+def load_token_info_func(scheme: SecurityScheme) -> TokenInfoFunc:
+    if not scheme.x_token_info_func:
+        raise SecurityException("Missing token info function")
+    try:
+        return load_handler(scheme.x_token_info_func)
+    except ValueError as err:
+        raise SecurityException from err
+
+
+def load_api_key_info_func(scheme: SecurityScheme) -> APIKeyInfoFunc:
+    if not scheme.x_api_key_info_func:
+        raise SecurityException("Missing API Key info function")
+    try:
+        return load_handler(scheme.x_api_key_info_func)
+    except ValueError as err:
+        raise SecurityException from err
+
+
+def load_bearer_info_func(scheme: SecurityScheme) -> BearerInfoFunc:
+    if not scheme.x_bearer_info_func:
+        raise SecurityException("Missing Bearer info function")
+    try:
+        return load_handler(scheme.x_bearer_info_func)
+    except ValueError as err:
+        raise SecurityException from err
+
+
+def validate_token_info(
+    token_info: InternalSecurityCheckResponse,
+    scope_validate_func: ScopeValidateFunc,
+    required_scopes: Sequence[str],
+) -> InternalSecurityCheckResponse:
+    scopes: Optional[Sequence[str]] = None
+    if not token_info:
+        return None
+
+    if "scopes" in token_info:
+        scopes = token_info.get("scopes")
+    elif "scope" in token_info:
+        scope = token_info.get("scope")
+        if isinstance(scope, str):
+            scopes = scope.split()
+        else:
+            raise ValueError("'scope' should be a string")
+
+    if not scopes:
+        raise ValueError("missing scopes in token info")
+
+    if not scope_validate_func(required_scopes, scopes):
+        raise SecurityException(
+            f"Invalid scopes: required: {required_scopes}, provided: {scopes}"
+        )
+
+    return token_info
+
+
+def build_http_security_check(
+    requirement: InternalSecurityRequirement, scheme: SecurityScheme
+) -> Optional[InternalSecurityCheck]:
+    _, required_scopes = requirement
+
+    if scheme.scheme == HTTPAuthenticationScheme.BASIC:
+        basic_info_func = load_basic_info_func(scheme)
+
+        def http_security_check(request: Request) -> InternalSecurityCheckResponse:
+            return validate_basic(request, basic_info_func, required_scopes)
+
+        return http_security_check
```

Comment:
```suggestion
        return partial(validate_basic, request, basic_info_func, required_scopes)
```
using [functools.partial](https://docs.python.org/3/library/functools.html#functools.partial)